ui/src/app/views/application/show/workflow/application.workflow.component.ts

```diff
@@ -193,29 +191,39 @@ export class ApplicationWorkflowComponent implements OnInit {
      * Action when changing branch
      */
     changeBranch(): void {
-        // reinit verison filter
-        this.applicationFilter.version = '';
-        jQuery('.cdsVersion div.text')[0].textContent = '';
-        this.changeVersion();
-
         // Load the versions of the new branch
-        this.loadVersions(this.project.key, this.application.name);
+        this.loadVersions(this.project.key, this.application.name)
+            .subscribe(() => this.changeVersion());
     };
 
     /**
      * Action when changing version
      */
-    changeVersion(): void {
+    changeVersion(version?: number): void {
         this.applicationFilter.branch = this.applicationFilter.branch.trim();
-        if (this.applicationFilter.version.trim() === '') {
-            this.applicationFilter.version = 0;
+
+        if (!version && Array.isArray(this.versions) && this.versions.length) {
+            this.applicationFilter.version = this.versions[1];
+        }
+
+        if (version) {
+            this.applicationFilter.version = version;
         }
+
         this._router.navigate(['/project/', this.project.key, 'application', this.application.name],
             {queryParams: {tab: 'workflow', branch: this.applicationFilter.branch, version: this.applicationFilter.version}});
         this.changeWorkerEvent.emit(true);
         this.clearTree(this.application.workflows);
     }
 
+    /**
+     * Load the list of version for the current application on the selected branch
+     */
+    loadVersions(key: string, appName: string): Observable<Array<number>> {
+        return this._appWorkflow.getVersions(key, appName, this.applicationFilter.branch)
+            .map((versions) => this.versions = versions);
+    };
+
```

Comment:
1. In fact branch is mandatory, there is no sense to mix workflow from different branch.
Default branch filter has to be default branch from git repository
2. If you remove empty version, you must had something to remove the filter